### ISSUE TITLE
Suggested method to authenticate against sandbox / test environment?

### ISSUE THREAD

**Issue by aaronbauman** [Lusitanian/PHPoAuthLib#542]
I see many of the OAuth Service providers have hard-coded endpoints for production services.
e.g. login.salesforce.com, www.paypal.com, etc.

For my application, I need flexibility to set the endpoint base URL differently, based on my SLDC. What's the recommended way to do this? Just extend the class and override those methods?

If that's the recommendation, would you consider a patch to include a more robust framework to swap out the endpoints? Or is this overkill?

Thanks for your work on this.

**Comment by Lusitanian** [comment]
Yeah honestly with the way the code structure is right now your best bet is extending the class and overriding the methods, naming the derived class something indicating it targets a different environment.

If you have a proposal for making it cleaner let me know
